### PR TITLE
refactor: update `blas/ext/base/dasumpw` to follow current project conventions

### LINKED ISSUE #1470: [RFC]: refactor `blas/ext/base/dasumpw` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dasumpw`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dasumpw) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by rajutkarsh07**
Resolves #1470 

## Description
update blas/ext/base/dasumpw to follow current project conventions

> What is the purpose of this pull request?

This pull request:

-  progresses #1152 
-  fixes #1470 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by rajutkarsh07** [comment]
@Planeshifter please review this PR

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2020 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/README.md]: We don't update the copyright years for existing files.

```suggestion
Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/benchmark/benchmark.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/benchmark/benchmark.native.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/benchmark/benchmark.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/lib/ndarray.native.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/lib/ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/lib/ndarray.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/lib/ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/lib/index.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/lib/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/manifest.json [hunk 7]
```diff
@@ -1,40 +1,45 @@
...
-				"./include"
-			],
-			"libraries": [
-				"-lm"
-			],
-			"libpath": [],
-			"dependencies": []
-		}
-	]
+    "options": {},
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/manifest.json]: This file needs 2-space indentation and we should add configurations for examples and benchmarks, see the linked PRs in the RFC.

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/docs/repl.txt [hunk 8]
```diff
@@ -92,4 +92,3 @@
 
     See Also
     --------
-
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/docs/repl.txt]: Removal of this line change should be reverted. 

I believe there may be some updates required for the `repl.txt` file as part of this refactoring, though.

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/examples/index.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/examples/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/benchmark/benchmark.ndarray.native.js [hunk 10]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/benchmark/benchmark.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/benchmark/benchmark.ndarray.js [hunk 11]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/benchmark/benchmark.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/src/addon.c [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+*
+* ## Notes
+*
+* -   When called from JavaScript, the function expects three arguments:
+*
+*     -   `N`: number of indexed elements
+*     -   `X`: input array
+*     -   `stride`: stride length
+*/
+napi_value node_dasumpw( napi_env env, napi_callback_info info ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/src/addon.c]: This file should be simplified as part of this refactoring to make use of the available macros, please consult the linked PRs for how this should look like.

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/src/addon.c [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
+#include "stdlib/blas/ext/base/dasumpw.h"
+#include <node_api.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <assert.h>
+
+/**
+* Computes the sum of absolute values (L1 norm) of double-precision floating-point strided array elements using pairwise summation.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/src/addon.c]: This comment should be replaced as part of the refactoring.

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/test/test.dasumpw.js [hunk 14]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/test/test.dasumpw.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/test/test.dasumpw.native.js [hunk 15]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/test/test.dasumpw.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/test/test.ndarray.js [hunk 16]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/test/test.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/test/test.ndarray.native.js [hunk 17]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/test/test.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks for your PR!

The main changes needed are updating the `addon.c` file and the `manifest.json` in order to get it ready for merging.

**Comment by rajutkarsh07** [comment]
@Planeshifter I have made the suggested changes

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/README.md [hunk 18]
```diff
@@ -154,18 +147,11 @@ var v = dasumpw.ndarray( N, x, 2, 1 );
...
 var dasumpw = require( '@stdlib/blas/ext/base/dasumpw' );
 
-var x;
-var i;
-
-x = new Float64Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = round( randu()*100.0 );
-}
+var x = filledarrayBy(10, 'float64', discreteUniform(0, 100));
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float64', discreteUniform( 0, 100 ) );
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
@rajutkarsh07 CI checks are currently failing. Looks like we need an additional empty line at L51 of `repl.txt` and I think `c_dasum` in the `addon.c` file has to be replaced with `stdlib_strided_dasumpw` for tests to pass. I also believe some changes to `manifest.json` are still outstanding. Hopefully fixing up these issues won't be too burdensome. Thanks for working on this PR!

**Comment by rajutkarsh07** [comment]
@Planeshifter made the changes in manifest.json and addon.c

File: lib/node_modules/@stdlib/blas/ext/base/dasumpw/src/addon.c [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, strideX, argv, 1 );
+
+	napi_value v;
+	napi_status status = napi_create_double( env, stdlib_strided_dasumpw( N, (double *)X, strideX ), &v );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dasumpw/src/addon.c]: No need for this cast anymore:

```suggestion
	napi_status status = napi_create_double( env, stdlib_strided_dasumpw( N, X, strideX ), &v );
```

**Review by Planeshifter — APPROVED** [review]
After a few additional changes to the `manifest.json`, CI checks are now passing. Will merge shortly. Thank you @rajutkarsh07 for your contribution!

**Pull request merged**